addons/account_edi_ubl_cii/data/ubl_20_templates.xml

```diff
@@ -0,0 +1,410 @@
+                    <cac:TaxCategory>
+                        <t t-call="{{TaxCategoryType_template}}">
+                            <t t-set="vals" t-value="foreach_vals.get('tax_category_vals', {})"/>
+                        </t>
+                    </cac:TaxCategory>
+                </cac:TaxSubtotal>
+            </t>
+        </t>
+    </template>
+
+    <template id="ubl_20_AllowanceChargeType">
+        <t xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2"
+           xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2"
+           xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
+            <cbc:ChargeIndicator t-out="vals.get('charge_indicator')"/>
+            <cbc:AllowanceChargeReasonCode t-out="vals.get('allowance_charge_reason_code')"/>
+            <cbc:AllowanceChargeReason t-out="vals.get('allowance_charge_reason')"/>
+            <cbc:MultiplierFactorNumeric t-out="vals.get('multiplier_factor')"/>
+            <cbc:Amount
+                    t-att-currencyID="vals.get('currency_name')"
+                    t-out="format_float(vals.get('amount'), vals.get('currency_dp'))"/>
+            <t t-foreach="vals.get('tax_category_vals', [])" t-as="foreach_vals">
+                <cac:TaxCategory>
+                    <t t-call="{{TaxCategoryType_template}}">
+                        <t t-set="vals" t-value="foreach_vals"/>
+                    </t>
+                </cac:TaxCategory>
+            </t>
+        </t>
+    </template>
+
+    <template id="ubl_20_InvoiceLineType">
+        <t xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2"
+           xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2"
+           xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
+            <cbc:ID t-out="vals.get('id')"/>
+            <cbc:Note t-out="vals.get('note')"/>
+            <cbc:InvoicedQuantity
+                    t-if="invoice.move_type == 'out_invoice'"
+                    t-att="vals.get('invoiced_quantity_attrs', {})"
+                    t-out="vals.get('invoiced_quantity')"/>
+            <cbc:CreditedQuantity
+                    t-if="invoice.move_type == 'out_refund'"
+                    t-att="vals.get('invoiced_quantity_attrs', {})"
+                    t-out="vals.get('invoiced_quantity')"/>
+            <cbc:LineExtensionAmount
+                    t-att-currencyID="vals['currency'].name"
+                    t-out="format_float(vals.get('line_extension_amount'), vals.get('currency_dp'))"/>
+            <!-- AllowanceCharge is only present for InvoiceLine, not for CreditNoteLine in UBL 2.0
+            (they are both present in UBL 2.1 and next)
+             http://www.datypic.com/sc/ubl20/e-cac_CreditNoteLine.html
+             http://www.datypic.com/sc/ubl20/e-cac_InvoiceLine.html
+             -->
+            <t t-if="invoice.move_type == 'out_invoice'">
```

Comment:
If I understand your idea correctly, If I do that, we'll have:
1. AllowanceCharge
2. TaxTotal 
for credit notes and invoices in UBL 2.1. It would be wrong since the order should be swapped for credit notes.